lib/queuePopulator/LogReader.js

```diff
@@ -290,14 +290,22 @@ class LogReader {
             timeoutMs: params.timeoutMs,
             logger: this.log.newRequestLogger(),
         };
-
+        const startTime = Date.now();
         async.waterfall([
             next => this._processReadRecords(params, batchState, next),
             next => this._processPrepareEntries(batchState, next),
             next => this._processPublishEntries(batchState, next),
             next => this._processSaveLogOffset(batchState, next),
         ],
             err => {
+                if (this._metricsHandler && this._metricsHandler.processingLag) {
```

Comment:
not sure if/how we should handle errors here : not sure if/when errors are expected, but it may very well be that either we don't care about duration in case of error OR we want to at least differentiate the duration of errors...

the easy way out is to just add some sort of 'result' label, so that we can get either the aggregated, succesful or failed metrics.